### ISSUE TITLE
Weird error messages when I miss to define macro name.

### ISSUE THREAD

**Issue by bryantalexander513** [rust-lang/rust#127612]
### Code

```Rust
macro_rules! {
    () => {};
}
```

### Current output

```Shell
error: cannot find macro `macro_rules` in this scope
 --> src/lib.rs:1:1
  |
1 | macro_rules! {
  | ^^^^^^^^^^^
  |
```

### Desired output

```Shell
error: missing name for `macro_rules!`
 --> src/lib.rs:1:1
  |
1 | macro_rules! {
  | ^^^^^^^^^^^
  |
```

### Rationale and extra context

I am wondering if I should import something to use `macro_rules!` inside lib.rs.
The error output shows that rust compiler doesn't recognize `macro_rules!` macro so it should be imported from anywhere else.

### Rust Version

```Shell
rustc 1.79.0 (129f3b996 2024-06-10)
binary: rustc
commit-hash: 129f3b9964af4d4a709d1383930ade12dfe7c081
commit-date: 2024-06-10
host: x86_64-pc-windows-msvc
release: 1.79.0
LLVM version: 18.1.7
```

**Comment by veera-sivarajan** [comment]
Fixed in #118317 and #127557

**Comment by fmease** [comment]
Duplicate of #126694
